jlibra-core/src/main/java/dev/jlibra/admissioncontrol/query/Event.java

```diff
@@ -22,15 +23,27 @@
 
     long getSequenceNumber();
 
+    ByteSequence getMetadata();
+
     static Event fromGrpcObject(types.Events.Event event) {
         byte[] eventData = event.getEventData().toByteArray();
         try (DataInputStream eventDataStream = new DataInputStream(new ByteArrayInputStream(eventData))) {
             long amount = Deserialization.readLong(eventDataStream, 8);
             ByteSequence address = Deserialization.readByteSequence(eventDataStream, 32);
+            ByteSequence message = ByteSequence.from(new byte[0]);
+            if (eventDataStream.available() > 0) {
+                ByteSequence messageData = Deserialization.readByteSequence(eventDataStream, eventDataStream.available());
+                boolean onlyZerosStream = IntStream.range(0, messageData.toArray().length).allMatch(i -> messageData.toArray()[i] == 0);
+                if (!onlyZerosStream) {
```

Comment:
In the libra serialization format, for a byte array, the first four bytes are the length of the data. In case where there is no metadata, there are just the bytes telling that the length is 0. The Deserialization.readByteSequence will check part that and return an empty array in this case, meaning the data does not contain zeroes when there's no metadata. 

- As a change, I suggest you can remove this onlyZeroStream check and just put the value of the byte sequence to the metadata attribute.

- I also feel it would be natural to have the metadata field in Event as Optional here, then it would be easy to check if the event contains metadata and the developer would not need to check for null or empty array

Reply:
Actually that advice was a bit incorrect, you need to read the size first and then the data, like this:
``
int length = readInt(in, 4);
ByteSequence data = readByteSequence(in, length);
``

Reply:
Wow, now I get it - couldn't wrap my head around those 4 zeros. Will change this.